scripts/nginx/readarr.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+# Nginx conf for Readarr
+# Bakerboy448 for swizzin 2020
+# Based on Radarr v3 installationn by flying sausages
+master=$(cut -d: -f1 < /root/.master.info)
+
+cat > /etc/nginx/apps/readarr.conf << readarr
+location /readarr {
+  proxy_pass        http://127.0.0.1:8787/readarr;
+  proxy_set_header Host \$proxy_host;
+  proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
+  proxy_set_header X-Forwarded-Proto \$scheme;
+  proxy_redirect off;
+  auth_basic "What's the password?";
+  auth_basic_user_file /etc/htpasswd.d/htpasswd.${master};
+
+  proxy_http_version 1.1;
+  proxy_set_header Upgrade \$http_upgrade;
+  proxy_set_header Connection \$http_connection;
+}
+readarr
+
+isactive=$(systemctl is-active readarr)
+
+if [[ $isactive == "active" ]]; then
+    echo_log_only "Stopping readarr"
+    systemctl stop readarr
+fi
+user=$(grep User /etc/systemd/system/readarr.service | cut -d= -f2)
+echo_log_only "readarr user detected as $user"
+apikey=$(grep -oPm1 "(?<=<ApiKey>)[^<]+" /home/"$user"/.config/readarr/config.xml)
+echo_log_only "Apikey = $apikey" >> "$log"
+#TODO change Branch whenever that becomes relevant
+cat > /home/"$user"/.config/readarr/config.xml << readarr
+# Readarr is Pre-Alpha, set logs to trace
+<Config>
+  <LogLevel>trace</LogLevel>
+  <UpdateMechanism>BuiltIn</UpdateMechanism>
+  <BindAddress>127.0.0.1</BindAddress>
+  <Port>8787</Port>
+  <SslPort>8788</SslPort>
+  <EnableSsl>False</EnableSsl>
+  <LaunchBrowser>False</LaunchBrowser>
+  <ApiKey>${apikey}</ApiKey>
+  <AuthenticationMethod>None</AuthenticationMethod>
+  <UrlBase>readarr</UrlBase>
+</Config>
+readarr
+
+chown -R "$user":"$user" /home/"$user"/.config/readarr
+
+#shellcheck source=sources/functions/utils
+. /etc/swizzin/sources/functions/utils
+systemctl start readarr -q # Switch readarr on regardless whether it was off before or not as we need to have it online to trigger this cahnge
+if ! timeout 15 bash -c -- "while ! curl -fL \"http://127.0.0.1:8787/api/v3/system/status?apiKey=${apikey}\" >> \"$log\" 2>&1; do sleep 5; done"; then
+    echo_error "readarr API did not respond as expected. Please make sure readarr is on v3 and running."
+    exit 1
+else
+    urlbase="$(curl -sL "http://127.0.0.1:8787/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
+    echo_log_only "readarr API tested and reachable"
+fi
+
+payload=$(curl -sL "http://127.0.0.1:8787/api/v3/config/host?apikey=${apikey}" | jq ".certificateValidation = \"disabledForLocalAddresses\"")
+echo_log_only "Payload = \n${payload}"
```

Comment:
This should be either removed, or the api key and possibly passwords should be removed from this later on when the `sysinfo` PR is merged

Reply:
Punt to sysinfo